cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/anomaly/MultivariateAnomalyDetection.scala

```diff
@@ -435,16 +444,19 @@ class FitMultivariateAnomaly(override val uid: String) extends Estimator[DetectM
 
   def getDisplayName: String = $(displayName)
 
-  protected def prepareEntity(source: String): Option[AbstractHttpEntity] = {
+  setDefault(slidingWindow -> 300, alignMode -> "Outer", fillNAMethod -> "Linear")
+
+  protected def prepareEntity(dataSource: String): Option[AbstractHttpEntity] = {
     Some(new StringEntity(
       MAERequest(
-        source,
+        dataSource,
+        dataSchema,
         getStartTime,
         getEndTime,
-        get(slidingWindow),
+        Option(getOrDefault(slidingWindow)),
```

Comment:
This won’t have the same semantics getOrDefault will throw an error if it’s not defined, take a look at getOrDefaults implementation and you might be able to slightly modify it to return an option instead